tests/io/hdf5_test.py

```diff
@@ -276,6 +279,26 @@ def test_dataset_with_many_coords():
         assert size1 < (len(ds1.coords) + 1) * rows * 8 + extra
 
 
+def test_data_group():
+    dg = sc.DataGroup({
+        'vector':
+        vector,
+        'data_array':
+        array_2d,
+        'dataset':
+        sc.Dataset({
+            'a': array_1d,
+            'b': array_2d
+        }),
+        'data group':
+        sc.DataGroup({
+            'v.same': vector,
+            'm/copy': matrix.copy()
+        })
+    })
+    check_roundtrip(dg)
```

Comment:
Can you add a test for items that are not scipp objects? We should either skip them, or fail with a reasonable exception?